serializer.rst

```diff
@@ -1,224 +1,454 @@
+common formats used by HTTP applications:
+
+* :class:`Symfony\\Component\\Serializer\\Encoder\\JsonEncoder`
+* :class:`Symfony\\Component\\Serializer\\Encoder\\XmlEncoder`
+* :class:`Symfony\\Component\\Serializer\\Encoder\\CsvEncoder`
+* :class:`Symfony\\Component\\Serializer\\Encoder\\YamlEncoder`
+
+Read more about these encoders and their configuration in
+:doc:`/serializer/encoders`.
+
+.. tip::
+
+    The `API Platform`_ project provides encoders for more advanced
+    formats:
+
+    * `JSON-LD`_ along with the `Hydra Core Vocabulary`_
+    * `OpenAPI`_ v2 (formerly Swagger) and v3
+    * `GraphQL`_
+    * `JSON:API`_
+    * `HAL`_
+
+.. _serializer-context:
+
+Serializer Context
+~~~~~~~~~~~~~~~~~~
+
+The serializer, and its normalizers and encoders, are configured through
+the *serializer context*. This context can be configured in multiple
+places:
+
+* `Globally through the framework configuration <Configure a Default Context>`_
+* `While serializing/deserializing <Pass Context while Serializing/Deserializing>`_
+* `For a specific property <Configure Context on a Specific Property>`_
+
+You can use all three options at the same time. When the same setting is
+configured in multiple places, the latter in the list above will override
+the previous one (e.g. the setting on a specific property overrides the one
+configured globally).
+
+Configure a Default Context
+...........................
+
+You can configure a default context in the framework configuration, for
+instance to disallow extra fields while deserializing:
 
 .. configuration-block::
 
     .. code-block:: yaml
 
-        # config/packages/framework.yaml
+        # config/packages/serializer.yaml
         framework:
-            # ...
             serializer:
                 default_context:
-                    enable_max_depth: true
+                    allow_extra_attributes: false
 
     .. code-block:: xml
 
-        <!-- config/packages/framework.xml -->
-        <framework:config>
-            <!-- ... -->
-            <framework:serializer>
-                <default-context enable-max-depth="true"/>
-            </framework:serializer>
-        </framework:config>
+        <!-- config/packages/serializer.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:serializer>
+                    <framework:default-context
+                        allow-extra-attributes="false"
+                    />
```

Comment:
looking at the XSD it should probably look like this:

```xml
<framework:default-context>
    <framework:allow-extra-attributes>false</framework:allow-extra-attributes>
</framework:default-context>

Reply:
That's unfortunate, the parser allows both and attributes for scalars is much nicer to read imho. But I'll change it to conform xsd :+1: